From 9e7b890a22507f06a34ed7f7956fb02392348187 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wim.taymans@collabora.co.uk>
Date: Thu, 23 Sep 2010 14:46:31 +0000
Subject: flvdemux: parse and use cts

For H264, there is an extra header containing the CTS, which is a
timestamp offset that should be applied to the PTS.
Parse this value and use it to adjust the pts. 
---
diff --git a/gst/flv/gstflvdemux.c b/gst/flv/gstflvdemux.c
index 4049939..e22eca7 100644
--- a/gst/flv/gstflvdemux.c
+++ b/gst/flv/gstflvdemux.c
@@ -1084,7 +1084,16 @@ gst_flv_demux_parse_tag_video (GstFlvDem
   if (codec_tag == 4 || codec_tag == 5) {
     codec_data = 2;
   } else if (codec_tag == 7) {
+    gint32 cts;
+
     codec_data = 5;
+
+    cts = GST_READ_UINT24_BE (data + 9);
+    cts = (cts + 0xff800000) ^ 0xff800000;
+
+    GST_LOG_OBJECT (demux, "got cts %d", cts);
+
+    pts = pts + cts;
   }
 
   GST_LOG_OBJECT (demux, "video tag with codec tag %u, keyframe (%d) "
--
cgit v0.8.3-6-g21f6
